src/core/security/auth.c

```diff
@@ -44,12 +44,15 @@
 #include "src/core/security/credentials.h"
 #include "src/core/surface/call.h"
 
+#define MAX_CREDENTIAL_METADATA_COUNT 4
```

Comment:
4 may be a little limiting... IAM is 2 already, any token is another one, that does not leave much for future stuff...